paper/src/main/java/com/github/igotyou/FactoryMod/factories/FurnCraftChestFactory.java

```diff
@@ -281,8 +281,9 @@ public void attemptToActivate(Player p, boolean onStartUp) {
 					setRecipe(autoRepair);
 				}
 			}
-			if (!hasInputMaterials() || (!rm.inDisrepair() && (currentRecipe instanceof  RepairRecipe))) {
```

Comment:
What was happening previously is that the repair recipe was being chosen in the previous code block, but then if the repair materials aren't available immediately, this if block evaluates to true and it flips back to the first recipe for which there are materials.